[action] Generate update_code_signing_settings action instead of deprecated automatic_code_signing

att55 opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

`automatic_code_signing` has `use_automatic_signing`, the key switches `Auto` to `Manual` code signing. And the defaulut value is `false`.

https://docs.fastlane.tools/actions/automatic_code_signing/#automatic_code_signing

Sample code is as below.

```
automatic_code_signing(
  path: "demo-project/demo/demo.xcodeproj",
  use_automatic_signing: false
)

automatic_code_signing(
  path: "demo-project/demo/demo.xcodeproj",
  use_automatic_signing: true
)
```

But, I feel the specifications below are strange and confusing users. 

1. Even though the action name is `automatic_code_signing`, the action allows to switch `Auto` to `Manual` code signing.
2. Even though the action name is `automatic_code_signing`, `Manual` code signing is executed by default because the default value of `use_automatic_signing` is `false`.

So, I added `code_signing` as alias of `automatic_code_signing` to prevent confusing users.

att55 committed:
add alias for code_signing

att55 committed:
add alias for automatic_code_signing

joshdholtz reviewed (changes requested):
@att55 👋 This is a very interesting change! I'm a little bit for but also a little bit against this 🤔 I do agree that this action should have a different name but I also don't want to just introduce another action/alias.

Aliases are nice (in this cases since some of the options conflict with the name of the action) but it makes things a little bit harder to maintain and harder to find other actions. 

If we are going to do this, I would rather move the logic from `AutomaticCodeSigning` into one that is better named and add a message saying that we are deprecating `AutomaticCodeSigning` in the future.

The `code_signing` name feels a bit too generic to me (I actually thought this PR was for something like _sigh_ or _cert_ at first). I think it would need to be something like `update_code_signing_settings`

Thoughts?

att55 commented:
@joshdholtz Thank you so much for your review!

> I would rather move the logic from AutomaticCodeSigning into one that is better named and add a message saying that we are deprecating AutomaticCodeSigning in the future.
> I think it would need to be something like update_code_signing_settings

I think your plan is very good and `update_code_signing_settings` is effective to resolve this iseue! 😸 
I will try to adopt your plan in this PR 👍

att55 committed:
adopt update_code_signing_settings action

att55 committed:
adopt update_code_signing_settings action

att55 committed:
adopt update_code_singing_settings action

att55 committed:
take deprecated message into account in test

att55 committed:
narrow down one category

att55 committed:
add test of update_code_signing_settings

att55 committed:
move a position of comma

att55 commented:
@joshdholtz I adopted our plan in this PR 👍Please check when you can 🙏 

The outline is below.
- added the message saying that we are deprecating automatic_code_signing in automatic_code_signing action .
- generated update_code_signing_setttings action instead of automatic_code_signing
- added a test file for update_code_signing_setttings
- added fixtures(xcodeproj) to exec the test of update_code_signing_setttings

joshdholtz commented on fastlane/spec/actions_specs/automatic_code_signing_spec.rb:
```diff
@@ -34,6 +34,8 @@
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
       expect(UI).to receive(:success).with("\t * today")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
```
```suggestion
      expect(UI).to receive(:deprecated).with("The `automatic_code_signing` action has been deprecated,")
```

joshdholtz commented on fastlane/spec/actions_specs/automatic_code_signing_spec.rb:
```diff
@@ -34,6 +34,8 @@
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
       expect(UI).to receive(:success).with("\t * today")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
+      expect(UI).to receive(:important).with("Please use `update_code_signing_settings` action instead.")
```
```suggestion
      expect(UI).to receive(:deprecated).with("Please use `update_code_signing_settings` action instead.")
```

joshdholtz commented on fastlane/spec/actions_specs/automatic_code_signing_spec.rb:
```diff
@@ -76,6 +78,8 @@
       # G3KGXDXQL9
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
```
```suggestion
      expect(UI).to receive(:deprecated).with("The `automatic_code_signing` action has been deprecated,")
```

joshdholtz commented on fastlane/spec/actions_specs/automatic_code_signing_spec.rb:
```diff
@@ -76,6 +78,8 @@
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
+      expect(UI).to receive(:important).with("Please use `update_code_signing_settings` action instead.")
```
```suggestion
      expect(UI).to receive(:deprecated).with("Please use `update_code_signing_settings` action instead.")
```

joshdholtz commented on fastlane/spec/actions_specs/automatic_code_signing_spec.rb:
```diff
@@ -106,6 +110,8 @@
       # G3KGXDXQL9
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
```
```suggestion
      expect(UI).to receive(:deprecated).with("The `automatic_code_signing` action has been deprecated,")
```

joshdholtz commented on fastlane/spec/actions_specs/automatic_code_signing_spec.rb:
```diff
@@ -106,6 +110,8 @@
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
+      expect(UI).to receive(:important).with("Please use `update_code_signing_settings` action instead.")
```
```suggestion
      expect(UI).to receive(:deprecated).with("Please use `update_code_signing_settings` action instead.")
```

joshdholtz commented on fastlane/spec/actions_specs/automatic_code_signing_spec.rb:
```diff
@@ -142,6 +148,8 @@
       # G3KGXDXQL9
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
```
```suggestion
      expect(UI).to receive(:deprecated).with("The `automatic_code_signing` action has been deprecated,")
```

joshdholtz commented on fastlane/spec/actions_specs/automatic_code_signing_spec.rb:
```diff
@@ -142,6 +148,8 @@
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
+      expect(UI).to receive(:important).with("Please use `update_code_signing_settings` action instead.")
```
```suggestion
      expect(UI).to receive(:deprecated).with("Please use `update_code_signing_settings` action instead.")
```

joshdholtz commented on fastlane/spec/actions_specs/automatic_code_signing_spec.rb:
```diff
@@ -174,6 +182,8 @@
       # G3KGXDXQL9
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
```
```suggestion
      expect(UI).to receive(:deprecated).with("The `automatic_code_signing` action has been deprecated,")
```

joshdholtz commented on fastlane/spec/actions_specs/automatic_code_signing_spec.rb:
```diff
@@ -174,6 +182,8 @@
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
+      expect(UI).to receive(:important).with("Please use `update_code_signing_settings` action instead.")
```
```suggestion
      expect(UI).to receive(:deprecated).with("Please use `update_code_signing_settings` action instead.")
```

joshdholtz commented on fastlane/spec/actions_specs/automatic_code_signing_spec.rb:
```diff
@@ -206,6 +216,8 @@
       # G3KGXDXQL9
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
```
```suggestion
      expect(UI).to receive(:deprecated).with("The `automatic_code_signing` action has been deprecated,")
```

joshdholtz commented on fastlane/spec/actions_specs/automatic_code_signing_spec.rb:
```diff
@@ -206,6 +216,8 @@
       allow(UI).to receive(:success)
       expect(UI).to receive(:success).with("Successfully updated project settings to use Code Sign Style = 'Manual'")
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
+      expect(UI).to receive(:important).with("Please use `update_code_signing_settings` action instead.")
```
```suggestion
      expect(UI).to receive(:deprecated).with("Please use `update_code_signing_settings` action instead.")
```

joshdholtz commented on fastlane/spec/actions_specs/automatic_code_signing_spec.rb:
```diff
@@ -236,6 +248,8 @@
 
     it "targets not found notice" do
       allow(UI).to receive(:important)
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
```
```suggestion
      expect(UI).to receive(:deprecated).with("The `automatic_code_signing` action has been deprecated,")
```

joshdholtz commented on fastlane/spec/actions_specs/automatic_code_signing_spec.rb:
```diff
@@ -236,6 +248,8 @@
     it "targets not found notice" do
       allow(UI).to receive(:important)
+      expect(UI).to receive(:important).with("The `automatic_code_signing` action has been deprecated,")
+      expect(UI).to receive(:important).with("Please use `update_code_signing_settings` action instead.")
```
```suggestion
      expect(UI).to receive(:deprecated).with("Please use `update_code_signing_settings` action instead.")
```

joshdholtz reviewed (changes requested):
Thanks SO much for making the changes I suggested 🥰 One more small change! If we could use `UI.deprecated` instead of `UI.important` that would be 💯

att55 committed:
use UI.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

use ui.deprecated

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

att55 commented:
@googlebot I consent.

att55 commented:
@googlebot I consent.

att55 committed:
generate update_code_signing_settings action

att55 commented:
@googlebot I consent.

att55 commented:
I thought I've already consented CLAs... 🤔??

att55 commented:
@joshdholtz Thank you so much for your review and requesting suggested changes to this PR!  
I fixed to use `UI.deprecated` instead of `UI.important`. 👍 
Sorry to bother you, but please check when you can 😄

att55 commented:
Sorry... I didn't understand how to pass "size-label" and "cla/google"...

att55 commented:
@joshdholtz Sorry to bother you... please check this PR when you can 🙏

joshdholtz commented:
Yo yo yo! Will check first thing tomorrow morning ❤️

att55 commented:
Thank you so much ❤️ It's file, when you really have time 😸

att55 commented:
@joshdholtz ( When you can... 👀🙏 )

joshdholtz commented:
Aye aye! Getting back to things first thing tomorrow morning 👌

joshdholtz commented:
@googlebot I consent.

joshdholtz reviewed (approved):
Looks good to me! Thanks for making those changes 😊 Appreciate all your contributions ❤️
